src/runtime/tests/refiner-test.ts

```diff
@@ -17,338 +17,362 @@ import {Schema} from '../schema.js';
+      particle Foo
+        input: reads Something {num: Boolean [ num == not (true or false) ] }
+    `);
+    const typeData = {'num': 'Boolean'};
+    const ref1 = Refinement.fromAst(manifestAst1[0].args[0].type.fields[0].type.refinement, typeData);
+    ref1.normalize();
+    const manifestAst2 = parse(`
+      particle Foo
+        input: reads Something {num: Boolean [ not num ] }
+    `);
+    const ref2 = Refinement.fromAst(manifestAst2[0].args[0].type.fields[0].type.refinement, typeData);
+    // normalized version of ref1 should be the same as ref2
+    assert.strictEqual(JSON.stringify(ref1), JSON.stringify(ref2));
+  }));
+  it('tests if primitive math expressions are automatically evaluated', Flags.withFieldRefinementsAllowed(async () => {
+    const manifestAst1 = parse(`
+      particle Foo
+        input: reads Something {num: Number [ (2+11-9) > num or False ] }
+    `);
+    const typeData = {'num': 'Number'};
+    const ref1 = Refinement.fromAst(manifestAst1[0].args[0].type.fields[0].type.refinement, typeData);
+    ref1.normalize();
+    const manifestAst2 = parse(`
+      particle Foo
+        input: reads Something {num: Number [ num < 4 ] }
+    `);
+    const ref2 = Refinement.fromAst(manifestAst2[0].args[0].type.fields[0].type.refinement, typeData);
+    // normalized version of ref1 should be the same as ref2
+    assert.strictEqual(JSON.stringify(ref1), JSON.stringify(ref2));
+  }));
+  it(`tests if multiple 'not's are cancelled. `, Flags.withFieldRefinementsAllowed(async () => {
+    const manifestAst1 = parse(`
+      particle Foo
+        input: reads Something {num: Boolean [ not (not num) ] }
+    `);
+    const typeData = {'num': 'Boolean'};
+    const ref1 = Refinement.fromAst(manifestAst1[0].args[0].type.fields[0].type.refinement, typeData);
+    ref1.normalize();
+    const manifestAst2 = parse(`
+        particle Foo
+            input: reads Something {num: Boolean [ num ] }
+    `);
+    const ref2 = Refinement.fromAst(manifestAst2[0].args[0].type.fields[0].type.refinement, typeData);
+    // normalized version of ref1 should be the same as ref2
+    assert.strictEqual(JSON.stringify(ref1), JSON.stringify(ref2));
+  }));
+  it(`tests if expressions are rearranged 1`, Flags.withFieldRefinementsAllowed(async () => {
+    const manifestAst1 = parse(`
+        particle Foo
+            input: reads Something {num: Number [ (num + 5) <= (2*num - 11) ] }
+    `);
+    const typeData = {'num': 'Number'};
+    const ref1 = Refinement.fromAst(manifestAst1[0].args[0].type.fields[0].type.refinement, typeData);
+    ref1.normalize();
+    const manifestAst2 = parse(`
+        particle Foo
+            input: reads Something {num: Number [ num > 16 or num == 16] }
+    `);
+    const ref2 = Refinement.fromAst(manifestAst2[0].args[0].type.fields[0].type.refinement, typeData);
+    // normalized version of ref1 should be the same as ref2
+    assert.strictEqual(JSON.stringify(ref1), JSON.stringify(ref2));
```

Comment:
nit: alternative: `deepStringEqual` -- then there may not be a need for stringify.

Reply:
Took a bit of messing with the javascript's zeros (both 0 and -0) but I got this working. 
Thanks